admin login and registration (Patrick's code)

davidli731 opened the pull request:
users can choose to sign up as student or admin, and login to page /admin or /dnd depending on whether the user is an admin (currently the login always redirects you to /admin because a backend boolean is_admin needs to be returned, so I made a PR on the backend. After PR is accepted, this login function should work properly). I think Patrick was using an old master branch which wasn't up-to-date with the current master, so I copied his code over to my branch.

davidli731 committed:
admin login and registration (Patrick's code)

users can choose to sign up as student or admin, and login to page /admin or /dnd depending on whether the user is an admin (currently the login always redirects you to /admin because a backend boolean is_admin needs to be returned, so I made a PR on the backend. After PR is accepted, this login function should work properly)

jsaji committed:
Merge branch 'master' into hook_admin_page

jsaji committed:
Minor UI changes

jsaji reviewed (changes requested):
Good work - just needs a bug fix where successful admin registration is acknowledged and then redirects (right now it says there's an error on the form and stays there). Could also use 'confirm password' field on the register pages. :)

jsaji committed:
Redirects if no id is found

davidli731 committed:
added confirm password field

both password and confirm password needs to be the same to register for both admin and student

jsaji committed:
Merge branch 'master' into hook_admin_page

jsaji committed:
Conflict fix

jsaji committed:
Students can't access admin page
